instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/PeerServiceResolverImpl.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter;
+
+import static java.util.Comparator.comparing;
+import static java.util.Comparator.naturalOrder;
+import static java.util.Comparator.nullsFirst;
+
+import java.io.Serializable;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+class PeerServiceResolverImpl implements PeerServiceResolver {
+
+  private static final Logger logger = Logger.getLogger(PeerServiceResolverImpl.class.getName());
+
+  private static final Map<String, Map<ServiceMatcher, String>> mapping = new ConcurrentHashMap<>();
+
+  private static final Comparator<ServiceMatcher> matcherComparator =
+      nullsFirst(
+          comparing(ServiceMatcher::getPort, nullsFirst(naturalOrder()))
+              .thenComparing(comparing(ServiceMatcher::getPath, nullsFirst(naturalOrder()))));
+
+  PeerServiceResolverImpl(Map<String, String> peerServiceMapping) {
+    peerServiceMapping.forEach(
+        (key, serviceName) -> {
+          try {
+            URI uri;
+            if (key.contains("://")) {
+              uri = new URI(key);
+            } else {
+              uri = new URI("http://" + key);
+            }
```

Comment:
Constructing an `URI` is expensive; it'd be better performance-wise to parse the string by hand (e.g. see the [`UrlParser`](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/reactor/reactor-netty/reactor-netty-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/reactornetty/v1_0/UrlParser.java) class that we have in couple places)